src/index.ts

```diff
@@ -155,11 +155,11 @@ const txHistory = async (req: Request, res: Response) => {
     const untilBlockNum = await askBlockNumByHash(referenceBestBlock);
     const afterBlockNum = await askBlockNumByTxHash(referenceTx );
 
-    if(untilBlockNum.kind === "error" && untilBlockNum.errMsg !== utils.errMsgs.noValue) {
+    if(untilBlockNum.kind === "error" && typeof referenceBestBlock !== "undefined"){
```

Comment:
> typeof referenceBestBlock !== "undefined"

I'm not sure this is the best way to handle it given that `referenceBestBlock` should always be given. Maybe throw a different error if it's undefined?

Reply:
you're right, should just check for `.kind === "error"`